util/hash/src/lib.rs

```diff
@@ -30,28 +32,37 @@ pub const KECCAK_NULL_RLP: H256 = H256( [0x56, 0xe8, 0x1f, 0x17, 0x1b, 0xcc, 0x5
 /// The KECCAK of the RLP encoding of empty list.
 pub const KECCAK_EMPTY_LIST_RLP: H256 = H256( [0x1d, 0xcc, 0x4d, 0xe8, 0xde, 0xc7, 0x5d, 0x7a, 0xab, 0x85, 0xb5, 0x67, 0xb6, 0xcc, 0xd4, 0x1a, 0xd3, 0x12, 0x45, 0x1b, 0x94, 0x8a, 0x74, 0x13, 0xf0, 0xa1, 0x42, 0xfd, 0x40, 0xd4, 0x93, 0x47] );
 
-extern {
-	/// Hashes input. Returns -1 if either out or input does not exist. Otherwise returns 0.
-	pub fn keccak_256(out: *mut u8, outlen: usize, input: *const u8, inputlen: usize) -> i32;
-	/// Hashes input. Returns -1 if either out or input does not exist. Otherwise returns 0.
-	pub fn keccak_512(out: *mut u8, outlen: usize, input: *const u8, inputlen: usize) -> i32;
-}
 
 pub fn keccak<T: AsRef<[u8]>>(s: T) -> H256 {
 	let mut result = [0u8; 32];
 	write_keccak(s, &mut result);
 	H256(result)
 }
 
-pub fn write_keccak<T: AsRef<[u8]>>(s: T, dest: &mut [u8]) {
-	let input = s.as_ref();
+pub fn keccak_256_unchecked(out: *mut u8, outlen: usize, input: *const u8, inputlen: usize) {
```

Comment:
function should be marked unsafe -- it unconditionally takes the given pointers and assumes they are from valid data.